disnake/ui/modal.py

```diff
@@ -232,7 +232,7 @@ async def _scheduled_task(self, interaction: ModalInteraction) -> None:
         finally:
             # if the interaction was responded to (no matter if in the callback or error handler),
             # the modal closed for the user and therefore can be removed from the store
-            if interaction.response._responded:
+            if interaction.response._response_type:
```

Comment:
```suggestion
            if interaction.response._response_type is not None:
```

Reply:
I initially had it that way but reverted it: https://github.com/DisnakeDev/disnake/pull/737/commits/7e3bcca2fb2f808646907013a4b0d6fdc034a402

Reply:
All enum value checks should use `if x is not None:` instead of `if x:`; if a response type 0 existed (tbf, it doesn't), a check without `is not None` would break due to #556, similar to this:
```py
>>> import disnake
>>> bool(disnake.VerificationLevel.none)
False
```

Reply:
reverted by force-pushing the commit away.